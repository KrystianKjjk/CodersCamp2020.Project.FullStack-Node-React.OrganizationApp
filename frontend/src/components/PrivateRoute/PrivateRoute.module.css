.container{
    background-color: #292929;
    display: flex;
    width:100%;
    flex: 1;
}