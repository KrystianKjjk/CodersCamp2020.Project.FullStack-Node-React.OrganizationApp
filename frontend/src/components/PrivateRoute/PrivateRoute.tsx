import React from "react";
import styles from "./PrivateRoute.module.css";
import { Route, RouteProps } from "react-router-dom";
import Menu from "../Menu";

export interface PrivateRouteProps extends RouteProps {}

const PrivateRoute: React.FC<PrivateRouteProps> = (props) => {

//To do - check if user is logged in 

  return (
    <div className={styles.container}>
      <Menu />
      <div className={styles.content}>
        <Route {...props} />
      </div>
    </div>
  );
};

export default PrivateRoute;
