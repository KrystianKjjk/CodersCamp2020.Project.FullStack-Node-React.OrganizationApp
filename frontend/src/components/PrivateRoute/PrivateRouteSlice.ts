import { createSlice, PayloadAction } from '@reduxjs/toolkit';
import { AppThunk, RootState } from '../../app/store';

interface PrivateRouteState {
  
}

const initialState: PrivateRouteState = {
  
};

export const privateRouteSlice = createSlice({
  name: 'privateRoute',
  initialState,
  reducers: {

  },
});

export const { } = privateRouteSlice.actions;

// if you want, add selectors here, change the one below, remember to register reducer in store.ts
// export const selectPrivateRoute = (state: RootState) => state.privateRoute.value;

export default privateRouteSlice.reducer;
