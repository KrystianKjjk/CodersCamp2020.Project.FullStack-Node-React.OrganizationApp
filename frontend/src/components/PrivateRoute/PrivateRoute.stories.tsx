import React from 'react';
import { Story, Meta } from '@storybook/react/types-6-0';
import PrivateRoute , { PrivateRouteProps } from './PrivateRoute';

export default {
  title: 'PrivateRoute component',
  component: PrivateRoute,
} as Meta;

const Template: Story<PrivateRouteProps> = (args) => <PrivateRoute {...args} />;

export const FirstInput = Template.bind({});
FirstInput.args = {
  // label: 'Write something: ',
};