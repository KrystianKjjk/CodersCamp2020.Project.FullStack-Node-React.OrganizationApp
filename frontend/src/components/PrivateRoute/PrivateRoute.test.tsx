import React from 'react';
import { render} from '@testing-library/react';
import PrivateRoute from './PrivateRoute';
import {BrowserRouter as Router} from 'react-router-dom'

describe('PrivateRoute', () => {
   it('renders properly', () => {
      const DummyComponent=()=>{
         return <h2>Dummy component</h2>
      }
      const {container}=render(<Router><PrivateRoute path="/"><DummyComponent/></PrivateRoute></Router>);
      expect(container).toMatchSnapshot();
   });
});
