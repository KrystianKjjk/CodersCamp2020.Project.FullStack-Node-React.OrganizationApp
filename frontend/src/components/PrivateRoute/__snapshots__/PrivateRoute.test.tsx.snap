// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivateRoute renders properly 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="makeStyles-root-1"
    >
      <nav
        class="MuiList-root MuiList-padding"
      >
        <div
          class="makeStyles-userDiv-3"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            style="padding-top: 20px; font-size: 40px;"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
            />
          </svg>
          <p
            style="font-weight: 500;"
          >
            Name Surname
          </p>
          <p>
            Admin
          </p>
        </div>
        <a
          aria-disabled="false"
          class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
          href="/users"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiListItemIcon-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16 11c1.66 0 2.99-1.34 2.99-3S17.66 5 16 5c-1.66 0-3 1.34-3 3s1.34 3 3 3zm-8 0c1.66 0 2.99-1.34 2.99-3S9.66 5 8 5C6.34 5 5 6.34 5 8s1.34 3 3 3zm0 2c-2.33 0-7 1.17-7 3.5V19h14v-2.5c0-2.33-4.67-3.5-7-3.5zm8 0c-.29 0-.62.02-.97.05 1.16.84 1.97 1.97 1.97 3.45V19h6v-2.5c0-2.33-4.67-3.5-7-3.5z"
              />
            </svg>
          </div>
          <div
            class="MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
            >
              Users
            </span>
          </div>
          <span
            class="MuiTouchRipple-root"
          />
        </a>
        <a
          aria-disabled="false"
          class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
          href="/courses"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiListItemIcon-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 22c1.1 0 2-.9 2-2h-4c0 1.1.89 2 2 2zm6-6v-5c0-3.07-1.64-5.64-4.5-6.32V4c0-.83-.67-1.5-1.5-1.5s-1.5.67-1.5 1.5v.68C7.63 5.36 6 7.92 6 11v5l-2 2v1h16v-1l-2-2z"
              />
            </svg>
          </div>
          <div
            class="MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
            >
              Courses
            </span>
          </div>
          <span
            class="MuiTouchRipple-root"
          />
        </a>
        <a
          aria-disabled="false"
          class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
          href="/sections"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiListItemIcon-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M4 8h4V4H4v4zm6 12h4v-4h-4v4zm-6 0h4v-4H4v4zm0-6h4v-4H4v4zm6 0h4v-4h-4v4zm6-10v4h4V4h-4zm-6 4h4V4h-4v4zm6 6h4v-4h-4v4zm0 6h4v-4h-4v4z"
              />
            </svg>
          </div>
          <div
            class="MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
            >
              Sections
            </span>
          </div>
          <span
            class="MuiTouchRipple-root"
          />
        </a>
        <a
          aria-disabled="false"
          class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
          href="/gradesheets"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiListItemIcon-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 3h-4.18C14.4 1.84 13.3 1 12 1c-1.3 0-2.4.84-2.82 2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-7 0c.55 0 1 .45 1 1s-.45 1-1 1-1-.45-1-1 .45-1 1-1zm2 14H7v-2h7v2zm3-4H7v-2h10v2zm0-4H7V7h10v2z"
              />
            </svg>
          </div>
          <div
            class="MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
            >
              Grade sheets
            </span>
          </div>
          <span
            class="MuiTouchRipple-root"
          />
        </a>
        <a
          aria-disabled="false"
          class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
          href="/projects"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiListItemIcon-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 3c-.46 0-.93.04-1.4.14-2.76.53-4.96 2.76-5.48 5.52-.48 2.61.48 5.01 2.22 6.56.43.38.66.91.66 1.47V19c0 1.1.9 2 2 2h.28c.35.6.98 1 1.72 1s1.38-.4 1.72-1H14c1.1 0 2-.9 2-2v-2.31c0-.55.22-1.09.64-1.46C18.09 13.95 19 12.08 19 10c0-3.87-3.13-7-7-7zm2 16h-4v-1h4v1zm0-2h-4v-1h4v1zm-1.5-5.59V14h-1v-2.59L9.67 9.59l.71-.71L12 10.5l1.62-1.62.71.71-1.83 1.82z"
              />
            </svg>
          </div>
          <div
            class="MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
            >
              Projects
            </span>
          </div>
          <span
            class="MuiTouchRipple-root"
          />
        </a>
        <a
          aria-disabled="false"
          class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
          href="/teamprojects"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiListItemIcon-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 3c-.46 0-.93.04-1.4.14-2.76.53-4.96 2.76-5.48 5.52-.48 2.61.48 5.01 2.22 6.56.43.38.66.91.66 1.47V19c0 1.1.9 2 2 2h.28c.35.6.98 1 1.72 1s1.38-.4 1.72-1H14c1.1 0 2-.9 2-2v-2.31c0-.55.22-1.09.64-1.46C18.09 13.95 19 12.08 19 10c0-3.87-3.13-7-7-7zm2 16h-4v-1h4v1zm0-2h-4v-1h4v1zm-1.5-5.59V14h-1v-2.59L9.67 9.59l.71-.71L12 10.5l1.62-1.62.71.71-1.83 1.82z"
              />
            </svg>
          </div>
          <div
            class="MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
            >
              Team projects
            </span>
          </div>
          <span
            class="MuiTouchRipple-root"
          />
        </a>
        <a
          aria-disabled="false"
          class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
          href="/teams"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiListItemIcon-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16 11c1.66 0 2.99-1.34 2.99-3S17.66 5 16 5c-1.66 0-3 1.34-3 3s1.34 3 3 3zm-8 0c1.66 0 2.99-1.34 2.99-3S9.66 5 8 5C6.34 5 5 6.34 5 8s1.34 3 3 3zm0 2c-2.33 0-7 1.17-7 3.5V19h14v-2.5c0-2.33-4.67-3.5-7-3.5zm8 0c-.29 0-.62.02-.97.05 1.16.84 1.97 1.97 1.97 3.45V19h6v-2.5c0-2.33-4.67-3.5-7-3.5z"
              />
            </svg>
          </div>
          <div
            class="MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
            >
              Teams
            </span>
          </div>
          <span
            class="MuiTouchRipple-root"
          />
        </a>
        <span
          class="makeStyles-span-2"
        >
          Settings
        </span>
        <a
          aria-disabled="false"
          class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
          href="/myprofile"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiListItemIcon-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19.14 12.94c.04-.3.06-.61.06-.94 0-.32-.02-.64-.07-.94l2.03-1.58c.18-.14.23-.41.12-.61l-1.92-3.32c-.12-.22-.37-.29-.59-.22l-2.39.96c-.5-.38-1.03-.7-1.62-.94l-.36-2.54c-.04-.24-.24-.41-.48-.41h-3.84c-.24 0-.43.17-.47.41l-.36 2.54c-.59.24-1.13.57-1.62.94l-2.39-.96c-.22-.08-.47 0-.59.22L2.74 8.87c-.12.21-.08.47.12.61l2.03 1.58c-.05.3-.09.63-.09.94s.02.64.07.94l-2.03 1.58c-.18.14-.23.41-.12.61l1.92 3.32c.12.22.37.29.59.22l2.39-.96c.5.38 1.03.7 1.62.94l.36 2.54c.05.24.24.41.48.41h3.84c.24 0 .44-.17.47-.41l.36-2.54c.59-.24 1.13-.56 1.62-.94l2.39.96c.22.08.47 0 .59-.22l1.92-3.32c.12-.22.07-.47-.12-.61l-2.01-1.58zM12 15.6c-1.98 0-3.6-1.62-3.6-3.6s1.62-3.6 3.6-3.6 3.6 1.62 3.6 3.6-1.62 3.6-3.6 3.6z"
              />
            </svg>
          </div>
          <div
            class="MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
            >
              My profile
            </span>
          </div>
          <span
            class="MuiTouchRipple-root"
          />
        </a>
      </nav>
    </div>
    <div
      class="content"
    >
      <h2>
        Dummy component
      </h2>
    </div>
  </div>
</div>
`;
